### ISSUE TITLE
A little question about the code

### ISSUE THREAD

**Issue by zhangruichi** [shijxcs/LIFT#9]
```python
（your code）
           template = "a photo of a {}."
            prompts = self.get_tokenized_prompts(classnames, template)
            text_features = self.model.encode_text(prompts)
            text_features = F.normalize(text_features, dim=-1)

        if cfg.backbone.startswith("CLIP-ViT"):
            text_features = text_features @ self.model.image_encoder.proj.t()
# 你使用了text_features @ self.model.image_encoder.proj.t() 
            text_features = F.normalize(text_features, dim=-1)
```

```python
（clip code visual_res_block ）
    def forward(self, x: torch.Tensor):
        x = self.conv1(x)  # shape = [*, width, grid, grid]
        x = x.reshape(x.shape[0], x.shape[1], -1)  # shape = [*, width, grid ** 2]
        x = x.permute(0, 2, 1)  # shape = [*, grid ** 2, width]
        x = torch.cat([self.class_embedding.to(x.dtype) + torch.zeros(x.shape[0], 1, x.shape[-1], dtype=x.dtype, device=x.device), x], dim=1)  # shape = [*, grid ** 2 + 1, width]
        x = x + self.positional_embedding.to(x.dtype)
        x = self.ln_pre(x)

        x = x.permute(1, 0, 2)  # NLD -> LND
        x = self.transformer(x)
        x = x.permute(1, 0, 2)  # LND -> NLD

        x = self.ln_post(x[:, 0, :])

        if self.proj is not None:
            x = x @ self.proj
# CLIP中使用的是 x @ self.proj
        return x
```
我有两个问题
1. 这里做text_features @ self.model.image_encoder.proj.t() 为什么，为什么要换到高维空间做匹配？
2. 况且如果使用了原参数升高到高维，应该右乘proj的逆矩阵，而不应该是转置吧，是不是有点不合理呢?（X@Y=Z  不能说明 X= Z@Y^T 吧）

如果您有时间，非常感谢您能为我解答！

**Comment by shijxcs** [comment]
你好，这主要是为了沿用 ViT 的架构作为 backbone，同时让 ViT+classifier 的模型具备 zero-shot CLIP 的能力。

Zero-shot CLIP 的推理过程可以简写为 logits = (ViT(x) @ proj) @ text_features.t()，其应当等价于 logits = ViT(x) @ (text_features @ proj.t()).t()。所以我们用 text_features @ proj.t() 作为 classifier 的初始化。

**Comment by zhangruichi** [comment]
好的，非常感谢您的解答
